hypervisor/lib/crypto/mbedtls/hkdf.c

```diff
@@ -80,102 +84,85 @@ int32_t mbedtls_hkdf_expand( const mbedtls_md_info_t *md, const uint8_t *prk,
-        return( MBEDTLS_ERR_HKDF_BAD_INPUT_DATA );
-    }
-
-    if( info == NULL )
-    {
-        info = (const uint8_t *) "";
-        info_len = 0;
-    }
-
-    n = okm_len / hash_len;
-
-    if( (okm_len % hash_len) != 0 )
-    {
-        n++;
-    }
-
-    /*
-     * Per RFC 5869 Section 2.3, okm_len must not exceed
-     * 255 times the hash length
-     */
-    if( n > 255 )
-    {
-        return( MBEDTLS_ERR_HKDF_BAD_INPUT_DATA );
-    }
-
-    mbedtls_md_init( &ctx );
-
-    if( (ret = mbedtls_md_setup( &ctx, md) ) != 0 )
-    {
-        goto exit;
-    }
+    if ( (okm == NULL) || (prk_len < hash_len) || (hash_len == 0U) ) {
+        ret = MBEDTLS_ERR_HKDF_BAD_INPUT_DATA;
+    } else {
 
-    /*
-     * Compute T = T(1) | T(2) | T(3) | ... | T(N)
-     * Where T(N) is defined in RFC 5869 Section 2.3
-     */
-    for( i = 1; i <= n; i++ )
-    {
-        size_t num_to_copy;
-        uint8_t c = i & 0xff;
-
-        ret = mbedtls_md_hmac_starts( &ctx, prk, prk_len );
-        if( ret != 0 )
-        {
-            goto exit;
+        if ( tmp_info == NULL ) {
+            tmp_info = (const uint8_t *) "";
+            tmp_info_len = 0U;
         }
 
-        ret = mbedtls_md_hmac_update( &ctx, t, t_len );
-        if( ret != 0 )
-        {
-            goto exit;
-        }
+        n = okm_len / hash_len;
 
-        ret = mbedtls_md_hmac_update( &ctx, info, info_len );
-        if( ret != 0 )
-        {
-            goto exit;
+        if ( (okm_len % hash_len) != 0U ) {
+            n++;
         }
 
-        /* The constant concatenated to the end of each T(n) is a single octet.
-         * */
-        ret = mbedtls_md_hmac_update( &ctx, &c, 1 );
-        if( ret != 0 )
-        {
-            goto exit;
+        /*
+         * Per RFC 5869 Section 2.3, okm_len must not exceed
+         * 255 times the hash length
+         */
+        if ( n > 255U ) {
+            ret = MBEDTLS_ERR_HKDF_BAD_INPUT_DATA;
+        } else {
+            mbedtls_md_init( &ctx );
+
+            ret = mbedtls_md_setup( &ctx, md);
+            if ( ret == 0 ) {
+
+                /*
+                 * Compute T = T(1) | T(2) | T(3) | ... | T(N)
+                 * Where T(N) is defined in RFC 5869 Section 2.3
+                 */
+                for ( i = 1; i <= n; i++ ) {
+                    size_t num_to_copy;
+                    uint8_t c = (uint8_t)(i & 0xffU);
+
+                    ret = mbedtls_md_hmac_starts( &ctx, prk, prk_len );
+                    if ( ret == 0 ) {
+                        ret = mbedtls_md_hmac_update( &ctx, t, t_len );
+                    }
+
+                    if ( ret == 0 ) {
+                        ret = mbedtls_md_hmac_update( &ctx, tmp_info, tmp_info_len );
+                    }
+
+                    /* The constant concatenated to the end of each T(n) is a single octet.
+                     * */
+                    if ( ret == 0 ) {
+                        ret = mbedtls_md_hmac_update( &ctx, &c, 1 );
```

Comment:
`1U`
Please check all the constant usages in your patches.
U for uint8_t, uint16_t, uint32_t, size_t
UL for uint64_t